### PR TITLE
Update sample creation scripts for groups sample 5-2

### PR DESCRIPTION
**Pull request by v-michaelmi**
## Purpose
Update the configure, cleanup and group creation scripts with sample 5-2

## Does this introduce a breaking change?

```
[ ] Yes
[X] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[X] Other... Please describe: Update to configuration scripts
```

## How to Test
* Smoke test sample using scripts in `AppCreationSripts` folder.

### CODE REVIEW

File: 5-WebApp-AuthZ/5-2-Groups/AppCreationScripts/sample.json [hunk 1]
```diff
@@ -7,7 +7,6 @@
     "RepositoryUrl": "microsoft-identity-platform-aspnetcore-webapp-tutorial",
     "Endpoint": "AAD v2.0"
   },
```
Comment (kalyankrishna1) [inline 5-WebApp-AuthZ/5-2-Groups/AppCreationScripts/sample.json:9]: Please update it to Basher standards

File: 5-WebApp-AuthZ/5-2-Groups/AppCreationScripts/sample.json [hunk 2]
```diff
@@ -22,10 +21,24 @@
...
+        }
+      ],
+      "SecurityGroups": [
+        {
+          "Name": "GroupAdmin",
+          "Description": "Admin Security Group"
+        },
+        {
+          "Name": "GroupMember",
+          "Description": "User Security Group"
```
Comment (derisen) [inline 5-WebApp-AuthZ/5-2-Groups/AppCreationScripts/sample.json:41]: These two manual steps below can be removed:

> {
>           "Comment": "On Azure Portal, create a security group named GroupAdmin, assign some users to it, and configure your ID and Access token to emit GroupID in your app registration. Configure the value for 'GroupAdmin' key in appsettings.json."
>         },
>         {
>           "Comment": "On Azure Portal, create a security group named GroupMember, assign some users to it, and configure your ID and Access token to emit GroupID in your app registration. Configure the value for 'GroupMember' key in appsettings.json."
>         }

**Comment by kalyankrishna1** [comment]
The graphhelper needs to use this one -> [https://github.com/Azure-Samples/ms-identity-javascript-angular-tutorial/blob/main/5-AccessControl/2-call-api-groups/API/TodoListAPI/Services/GraphHelper.cs](https://github.com/Azure-Samples/ms-identity-javascript-angular-tutorial/blob/main/5-AccessControl/2-call-api-groups/API/TodoListAPI/Services/GraphHelper.cs)

**Comment by kalyankrishna1** [comment]
The readme needs a comprehensive update to pick the latest from code gen

File: 5-WebApp-AuthZ/5-2-Groups/AppCreationScripts/sample.json [hunk 3]
```diff
@@ -5,9 +5,23 @@
...
+      "aspnet-core",
+      "azure-active-directory"
+    ],
+    "Library": [
+      "Microsoft.Identity.Web"
+    ],
+    "products": [
+      "azure-active-directory",
+      "ms-graph"
+    ]
```
Comment (derisen) [inline 5-WebApp-AuthZ/5-2-Groups/AppCreationScripts/sample.json:23]:
```suggestion
```

**Comment by kalyankrishna1** [comment]
Please copy the code and logic from [https://github.com/Azure-Samples/ms-identity-javascript-angular-tutorial/blob/main/5-AccessControl/2-call-api-groups/API/TodoListAPI](https://github.com/Azure-Samples/ms-identity-javascript-angular-tutorial/blob/main/5-AccessControl/2-call-api-groups/API/TodoListAPI)

File: 5-WebApp-AuthZ/5-2-Groups/AppCreationScripts/sample.json [hunk 4]
```diff
@@ -5,9 +5,23 @@
     "Client": "ASP.NET Core Web App",
     "Service": "Microsoft Graph",
     "RepositoryUrl": "microsoft-identity-platform-aspnetcore-webapp-tutorial",
-    "Endpoint": "AAD v2.0"
+    "Endpoint": "AAD v2.0",
+    "platform": "AspNetCore",
+    "Languages": [
+      "csharp"
+    ],
+    "Products": [
```
Comment (derisen) [inline 5-WebApp-AuthZ/5-2-Groups/AppCreationScripts/sample.json:13]: >     "Products": [

Shouldn't be an array. Can you compare with the header here:  https://raw.githubusercontent.com/Azure-Samples/ms-identity-javascript-angular-tutorial/main/5-AccessControl/2-call-api-groups/README.md

File: 5-WebApp-AuthZ/5-2-Groups/AppCreationScripts/sample.json [hunk 5]
```diff
@@ -5,9 +5,23 @@
...
+      "aspnet-core",
+      "azure-active-directory"
```
Comment (derisen) [inline 5-WebApp-AuthZ/5-2-Groups/AppCreationScripts/sample.json:15]:
```suggestion
      "azure-active-directory"
      "microsoft-identity-web"
```

Comment (derisen) [reply 5-WebApp-AuthZ/5-2-Groups/AppCreationScripts/sample.json:15]: `microsoft-identity-web` as it appears here: https://review.learn.microsoft.com/en-us/help/platform/metadata-taxonomies?branch=main

**Comment by kalyankrishna1** [comment]
More-- 
1. CAE challenge handling needs to be added in

File: 5-WebApp-AuthZ/5-2-Groups/README.md [hunk 6]
```diff
@@ -49,521 +71,385 @@ From your shell or command line:
...
+    1. Under **Supported account types**, select **Accounts in this organizational directory only**
+    1. Select **Register** to create the application.
+1. In the **Overview** blade, find and note the **Application (client) ID**. You use this value in your app's configuration file(s) later in your code.
+1. In the app's registration screen, select the **Authentication** blade to the left.
+1. If you don't have a platform added, select **Add a platform** and select the **Web** option.
+    1. In the **Redirect URI** section enter the following redirect URIs:
+        1. `https://localhost:44321/`
+        1. `https://localhost:44321/signin-oidc`
+    1. In the **Front-channel logout URL** section, set it to `https://localhost:44321/signout-oidc`.
+     1. In the **Implicit grant** section, check the **ID tokens** option as this sample requires the [Implicit grant flow](https://docs.microsoft.com/azure/active-directory/develop/v2-oauth2-implicit-grant-flow) to be enabled to sign-in the user and call an API.
```
Comment (derisen) [inline 5-WebApp-AuthZ/5-2-Groups/README.md:139]: >   1. In the **Implicit grant** section, check the **ID tokens** option as this sample requires the [Implicit grant flow](https://docs.microsoft.com/azure/active-directory/develop/v2-oauth2-implicit-grant-flow) to be enabled to sign-in the user and call an API.

This is a bit misleading as just checking the ID token is called hybrid flow, not implicit. If the codegen generates this block, we should probably update it? cc @kalyankrishna1

File: 5-WebApp-AuthZ/5-2-Groups/README.md [hunk 7]
```diff
@@ -49,521 +71,385 @@ From your shell or command line:
...
+    1. In the **Name** section, enter a meaningful application name that will be displayed to users of the app, for example `WebApp-GroupClaims`.
+    1. Under **Supported account types**, select **Accounts in this organizational directory only**
+    1. Select **Register** to create the application.
+1. In the **Overview** blade, find and note the **Application (client) ID**. You use this value in your app's configuration file(s) later in your code.
+1. In the app's registration screen, select the **Authentication** blade to the left.
+1. If you don't have a platform added, select **Add a platform** and select the **Web** option.
+    1. In the **Redirect URI** section enter the following redirect URIs:
+        1. `https://localhost:44321/`
+        1. `https://localhost:44321/signin-oidc`
+    1. In the **Front-channel logout URL** section, set it to `https://localhost:44321/signout-oidc`.
```
Comment (derisen) [inline 5-WebApp-AuthZ/5-2-Groups/README.md:138]:
```suggestion
    1. In the **Front-channel logout URL** section, set it to `https://localhost:44321/signout-callback-oidc`.
```

Comment (derisen) [reply 5-WebApp-AuthZ/5-2-Groups/README.md:138]: should match to what's in appsettings.json -also need to update sample.json

File: 5-WebApp-AuthZ/5-2-Groups/README.md [hunk 8]
```diff
@@ -49,521 +71,385 @@ From your shell or command line:
...
+##### Configure the webApp app (WebApp-GroupClaims) to use your app registration
+
+Open the project in your IDE (like Visual Studio or Visual Studio Code) to configure the code.
+
+> In the steps below, "ClientID" is the same as "Application ID" or "AppId".
+
+1. Open the `appsettings.json` file.
+1. Find the key `ClientId` and replace the existing value with the application ID (clientId) of `WebApp-GroupClaims` app copied from the Azure portal.
+1. Find the key `TenantId` and replace the existing value with your Azure AD tenant/directory ID.
+1. Find the key `Domain` and replace the existing value with your Azure AD tenant domain, ex. `contoso.onmicrosoft.com`.
```
Comment (derisen) [inline 5-WebApp-AuthZ/5-2-Groups/README.md:168]: I think we can omit **Domain** from here and appsettings.json

File: 5-WebApp-AuthZ/5-2-Groups/README.md [hunk 9]
```diff
@@ -49,521 +71,385 @@ From your shell or command line:
...
+1. Find the key `ClientSecret` and replace the existing value with the generated secret that you saved during the creation of `WebApp-GroupClaims` copied from the Azure portal.
+1. Find the key `Enter the objectID for GroupAdmin group copied from Azure Portal` and replace the existing value with $GroupAdmin.Id.
+1. Find the key `Enter the objectID for GroupMember group copied from Azure Portal` and replace the existing value with $GroupMember.Id.
+
+### Create Security Groups
```
Comment (derisen) [inline 5-WebApp-AuthZ/5-2-Groups/README.md:173]: This section should come before `Configure the webApp app (WebApp-GroupClaims) to use your app registration`

**Comment by derisen** [comment]
@v-michaelmi probably need to update the `README-incremental-instructions.md` as well

File: 5-WebApp-AuthZ/5-2-Groups/Controllers/UserProfileController.cs [hunk 10]
```diff
@@ -11,28 +15,52 @@ namespace WebApp_OpenIDConnect_DotNet.Controllers
...
+                    string claimChallenge = WwwAuthenticateParameters.GetClaimChallengeFromResponseHeaders(svcex.ResponseHeaders);
+                    _consentHandler.ChallengeUser(_graphScopes, claimChallenge);
+                    return new EmptyResult();
+                }
+                catch (Exception ex2)
+                {
+                    _consentHandler.HandleException(ex2);
+                }
+            }
+            catch (ServiceException svcex) when (svcex.Error.Code == "ImageNotFound")
```
Comment (kalyankrishna1) [inline 5-WebApp-AuthZ/5-2-Groups/Controllers/UserProfileController.cs:59]: >(ServiceException svcex) when (svcex.Error.Code == "ImageNotFound" [](http://example.com/codeflow?start=18&length=66)

thank you :)

**Review by kalyankrishna1 — APPROVED** [review]
:shipit:

File: 5-WebApp-AuthZ/5-2-Groups/README.md (hunk 8 above)
Comment (kalyankrishna1) [reply 5-WebApp-AuthZ/5-2-Groups/README.md:177]: +1

**Comment by v-michaelmi** [comment]
@kalyankrishna1 @derisen can we get rid of the `README-incremental-instructions.md` file? It doesn't seem to be relevant anymore.

File: 5-WebApp-AuthZ/5-2-Groups/README.md (hunk 6 above)
Comment (v-michaelmi) [reply 5-WebApp-AuthZ/5-2-Groups/README.md]: The issue is CodeGen prints out the instruction to enable implicit flow twice one time mentioning the `implicit flow` the other time just instructing to enable it.

I'll remove the duplicate but yes we'll need to come back to this.

File: 5-WebApp-AuthZ/5-2-Groups/AppCreationScripts/sample.json (hunk 4 above)
Comment (v-michaelmi) [reply 5-WebApp-AuthZ/5-2-Groups/AppCreationScripts/sample.json:13]: The `Products` property is registered as a list.

Is there something I'm missing?

File: 5-WebApp-AuthZ/5-2-Groups/README.md (hunk 9 above)
Comment (v-michaelmi) [reply 5-WebApp-AuthZ/5-2-Groups/README.md:182]: That's going to need a CodeGen update but I can swap it in for this sample and add it later.

**Pull request merged**